### PR TITLE
DragControls: Add `mouseButtons`

### PR DESCRIPTION
**Pull request by puxiao**
This PR introduces a new property `mouseButtons` to DragControls. 

This enhancement allows users to customize which mouse button can trigger drag interactions.

This makes it easy to use DragControls with other controllers.

<br>

For example: `DragControls + OrbitControls`

```diff
const myOrbitControls = new OrbitControls(...)

const myDragControls = new DragControls(...)
+ myDragControls.mouseButtons = { LEFT: MOUSE.DRAG, MIDDLE: null, RIGHT: null }

myDragControls.addEventListener('dragstart', () =>{
  myOrbitControls.enabled = false
})

myDragControls.addEventListener('dragend', () =>{
  myOrbitControls.enabled = true
})
```

<br>

I can achieve the following results:

* `MOUSE.LEFT (selected)` - drag mesh
* `MOUSE.LEFT (no selected)` - rotation of the camera
* `MOUSE.RIGH` - camera panning
* `MOUSE.MIDDLE` - camera zooming

### CODE REVIEW

**Comment by Mugen87** [comment]
What I don't like about your control classes is the fact that stuff like that is implemented differently in each module.

I think it's time to think about `THREE.Controls` which acts as an abstract upper class for all concrete controls. Mouse and keyboard configuration could be part of `THREE.Controls`. E.g.

```
class Controls extends EventDispatcher {

	constructor( object, domElement ) {

		super();

		this.object = object;
		this.domElement = domElement;

		this.enabled = true;

		this.keys = { LEFT: null, UP: null, RIGHT: null, BOTTOM: null };
		this.mouseButtons = { LEFT: null, MIDDLE: null, RIGHT: null };
		this.touches = { ONE: null, TWO: null };

	}

	connect() {}

	disconnect() {}

	dispose() {}

	update() {}

}
```
I don't think getters and setters are required though. Controls can predefine `keys`, `mouseButtons` and `touches` and the developer can overwrite these properties if required.

**Comment by puxiao** [comment]
> IMO, it's time to think about `THREE.Controls` which acts as an abstract upper class for all concrete controls. Mouse and keyboard configuration could be part of `THREE.Controls`. E.g.

I hope that day comes soon.

File: examples/jsm/controls/DragControls.js [hunk 1]
```diff
@@ -37,7 +37,7 @@ class DragControls extends EventDispatcher {
 
 		this.rotateSpeed = 1;
 
-		//
+		this.mouseButtons = { LEFT: true, MIDDLE: true, RIGHT: true };
```
Comment (Mugen87) [inline examples/jsm/controls/DragControls.js:40]: The idea is to have an object that maps buttons to actions. E.g. `LEFT: MOUSE.ROTATE`. 

For `DragControls`, I would expect something like this:
```js
this.mouseButtons = { LEFT: MOUSE.DRAG, MIDDLE: MOUSE.DRAG, RIGHT: MOUSE.DRAG };
```

Comment (Mugen87) [reply examples/jsm/controls/DragControls.js:40]: The actions of the `MOUSE` enum are maintained here: https://github.com/mrdoob/three.js/blob/5a7ef89952256ce7fb9ae3401f27c5f12ec4e276/src/constants.js#L3

**Comment by Mugen87** [comment]
I have refactored `DragControls` based on your PR. Would these changes work for you as well?

https://github.com/Mugen87/three.js/commit/db7c2f524858ad41c0bc5f255e17c89635262563

**Comment by puxiao** [comment]
Good job !!!

**Pull request closed without merging**